src/utils/external-provider.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * WARNING: This code is intended to be run both on the browser application and as standalone Node
+ * JS script application !
+ *
+ * These are helpers function for the import tool that are also used by the
+ * check-external-layers-providers.js script.
+ */
+
+function appendParamsToUrl(url, paramString) {
+    if (paramString) {
+        const parts = (url + ' ').split(/[?&]/)
+        url +=
+            parts.pop() === ' '
+                ? paramString
+                : parts.length > 0
+                ? '&' + paramString
+                : '?' + paramString
+    }
+    return url
+}
+
+/**
+ * Prepares URL for external layer upload
+ *
+ * @param url
+ * @param lang
+ * @returns {Promise<string>}
+ */
+export function transformUrl(url, lang) {
+    // If the url has no file extension or a map parameter,
+    // try to load a WMS/WMTS GetCapabilities.
+    if (
+        (!/\.(kml|kmz|xml|txt)/i.test(url) && !/\w+\/\w+\.[a-zA-Z]+$/i.test(url)) ||
+        /map=/i.test(url)
+    ) {
+        // Append WMS GetCapabilities default parameters
+        url = appendParamsToUrl(
+            url,
+            /wmts/i.test(url)
+                ? 'SERVICE=WMTS&REQUEST=GetCapabilities&VERSION=1.0.0'
+                : 'SERVICE=WMS&REQUEST=GetCapabilities&VERSION=1.3.0'
+        )
+
+        // Use lang param only for admin.ch servers
+        if (/admin\.ch/.test(url)) {
+            url = appendParamsToUrl(url, `lang=${lang}`)
+        }
+        // Replace the subdomain template if exists
+        url = url.replace(/{s}/, '')
+    }
+    // Save the good url for the import component.
+    return url
+}
+
+/**
+ * Check is provided string valid URL
+ *
+ * @param {string} urlToCheck
+ */
+export function isValidUrl(urlToCheck) {
+    let url
+
+    try {
+        url = new URL(urlToCheck)
+    } catch (_) {
+        return false
+    }
+
+    return url.protocol === 'http:' || url.protocol === 'https:'
```

Comment:
shouldn't we enforce `https://` usage by filtering out non HTTPS URLs? (or we follow the redirect to see if they force HTTPS on their end?)

Reply:
I would let redirect to work if they are any.